Tests/NIOTests/ByteBufferTest.swift

```diff
@@ -2134,3 +2204,11 @@ private func testReserveCapacityLarger_reallocHook(_ ptr: UnsafeMutableRawPointe
 private func testReserveCapacityLarger_memcpyHook(_ dst: UnsafeMutableRawPointer, _ src: UnsafeRawPointer, _ count: Int) -> Void {
     // No copying
 }
+
+extension ByteBuffer {
+    func assign(to pointer: inout UInt) {
```

Comment:
I think we should give this function a better name and also return a `UInt` instead of taking it `inout`.

So I think we could use

```
func storagePointerIntegerValue() -> UInt {
   ...
}
```